Tools/CISupport/ews-build/factories.py

```diff
@@ -342,3 +342,30 @@ def __init__(self, platform, configuration=None, architectures=None, additionalA
         self.addStep(Canonicalize())
         self.addStep(PushCommitToWebKitRepo())
         self.addStep(SetBuildSummary())
+
+
+class UnsafeMergeQueueFactory(factory.BuildFactory):
```

Comment:
can we avoid duplicating all these steps? Maybe have MergeQueueFactory as base class, and based on some class variables skip appropriate steps.